daemon/src/connection.rs

```diff
@@ -172,3 +175,49 @@ impl Actor {
 }
 
 impl xtra::Actor for Actor {}
+
+pub async fn connect(
+    mut maker_online_status_feed_receiver: watch::Receiver<ConnectionStatus>,
+    connection_actor_addr: xtra::Address<Actor>,
+    maker_identity_pk: x25519_dalek::PublicKey,
+    maker_addresses: Vec<SocketAddr>,
+) {
+    loop {
+        if maker_online_status_feed_receiver.borrow().clone() == ConnectionStatus::Offline {
```

Comment:
I would be nice if the reconnect logic _does not_ depend on this channel but is handled internally in the `connection::Actor`.

Reply:
Indeed, that would be nicer - I'll aim to submit a  refactor PR after this one (to minimize the merge conflicts and allow to test the functionality in the meantime)